src/mono/wasm/runtime/binding_support.js

```diff
@@ -218,7 +235,7 @@ var BindingSupportLib = {
 				this._mono_wasm_release_js_handle(thenable_js_handle);
 
 				// when FinalizationRegistry is not supported by this browser, we will do immediate cleanup after use
-				if (!this._js_owned_object_registry) {
+				if (!this._is_finalizarion_registry_supported) {
```

Comment:
I think if we're going to use it this way we should probably name it something like '_use_finalization_registry', so that it is less surprising to be able to flip it from true to false at runtime and have it "work".